### ISSUE TITLE
Resume training, how to let the learning rate drop on the original basis?

### ISSUE THREAD

**Issue by Ronglongzhou** [KaiyangZhou/deep-person-reid#270]
Hi, I was training my own dataset when I accidentally interrupted, and I continued training and found that the learning rate didn't drop at the break point, it dropped from the initial 0.0015.How can I solve this problem? I am looking forward to your reply.

**Comment by KaiyangZhou** [comment]
try this https://kaiyangzhou.github.io/deep-person-reid/user_guide.html#resume-training

if you are using `scripts/main.py`, have a look at this https://github.com/KaiyangZhou/deep-person-reid/blob/master/scripts/main.py#L172, what you need to do is to give `cfg.model.resume` a path which points to the model you saved

**Comment by ggalan87** [comment]
Almost in the same context: Isn't it purposeful to also store the lr_scheduler state in the checkpoint file?

**Comment by Ronglongzhou** [comment]
> 
> 
> try this https://kaiyangzhou.github.io/deep-person-reid/user_guide.html#resume-training
> 
> if you are using `scripts/main.py`, have a look at this https://github.com/KaiyangZhou/deep-person-reid/blob/master/scripts/main.py#L172, what you need to do is to give `cfg.model.resume` a path which points to the model you saved

Thank you very much for your reply， I have given the point path to cfg.model.resume to resume training successfully. However, I found that the learning rate did not drop according to the state before the interruption, but started a new drop pattern. Almost in the same context, is it better to preserve the lr_scheduler state in the checkpoint file?

**Comment by KaiyangZhou** [comment]
It could be a good idea. Will try this

**Comment by KaiyangZhou** [comment]
just updated the code to add scheduler to `resume_from_checkpoint`

it should work now (let me know if it fails)

and sorry for the delay

**Comment by Ronglongzhou** [comment]
> 
> 
> just updated the code to add scheduler to `resume_from_checkpoint`
> 
> it should work now (let me know if it fails)
> 
> and sorry for the delay

Good, Thank you for your hard work

**Comment by J-LINC** [comment]
> > 试试这个 https://kaiyangzhou.github.io/deep-person-reid/user_guide.html#resume-training
> > 如果您正在使用 ，请查看此 https://github.com/KaiyangZhou/deep-person-reid/blob/master/scripts/main.py#L172，您需要做的是给出一条指向您保存的模型的路径`scripts/main.py``cfg.model.resume`
> 
> 非常感谢您的回复，我已经给出了cfg.model.resume的点路径以成功恢复训练。但是，我发现学习率并没有按照中断前的状态下降，而是开始了新的下降模式。几乎在相同的上下文中，在检查点文件中保留lr_scheduler状态是否更好？

if you want to resume last lr, you must modify the code about  resume_from_checkpoint, that means `if optimizer is None ...`
